src/Analyser/NodeScopeResolver.php

```diff
@@ -308,12 +308,23 @@ public function processNodes(
 			}
 
 			$alreadyTerminated = true;
-			$nextStmt = $this->getFirstUnreachableNode(array_slice($nodes, $i + 1), true);
-			if (!$nextStmt instanceof Node\Stmt) {
+			$nextStmts = $this->getNextUnreachableStatements(array_slice($nodes, $i + 1), true);
+
+			if ($nextStmts === []) {
 				continue;
 			}
 
-			$nodeCallback(new UnreachableStatementNode($nextStmt), $scope);
+			$unreachableStatementNode = null;
+			foreach ($nextStmts as $nextStmt) {
+				if ($unreachableStatementNode instanceof UnreachableStatementNode) {
+					$unreachableStatementNode->addNextStatement($nextStmt);
```

Comment:
I generally want immutable objects. Please build the array first and then create the object.

Reply:
updated to immutable object https://github.com/phpstan/phpstan-src/commit/66596b45eb61df295e3db477ae99e4d45a48edff